lib/stripe/util.rb

```diff
@@ -119,21 +119,35 @@ def self.flatten_params_array(value, calculated_key)
     def self.parse_opts(opts)
       case opts
       when NilClass
```

Comment:
What do you think about just getting rid of this `when NilClass` clause altogether and letting the `else` clause handle it? (In which case you could also merge `raise_bad_api_key` into `check_bad_api_key`, since that would be the only place it's used.)